### DISCUSSION TITLE
Content creation failing

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Knowlg/Community#129] [Bugs]
We have created new channel as Channel creation failed while creation of the ORG in Lern.
https://github.com/orgs/Sunbird-Lern/discussions/125

But when the channel is created using Knowlg API, we have seen lot of default values being added for the channel. You can see below  properties assuming all the different primaryCategories added by default while creating the channel 
`contentPrimaryCategories`, `collectionPrimaryCategories`, `contentAdditionalCategories`, `collectionAdditionalCategories`

```
{
    "id": "api.channel.read",
    "ver": "1.0",
    "ts": "2023-05-26T11:51:53.562Z",
    "params": {
        "resmsgid": "b52fd3a0-fbbb-11ed-b5f1-a7f115ccf799",
        "msgid": "b5206a50-fbbb-11ed-8b1d-0f2cd1e1ec77",
        "status": "successful",
        "err": null,
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "channel": {
            "identifier": "sunbird",
            "lastStatusChangedOn": "2023-05-26T07:39:54.345+0000",
            "code": "sunbird",
            "defaultLicense": "CC BY 4.0",
            "consumerId": "f5f3c5a7-ae4b-4d09-abbe-048462f6fb6a",
            "description": "Default Organisation for Sunbird",
            "assetAdditionalCategories": [],
            "autoCreateBatch": "Enabled",
            "languageCode": [],
            "createdOn": "2023-05-26T07:39:54.345+0000",
            "objectType": "Channel",
            "versionKey": "1685087619709",
            "collectionPrimaryCategories": [
                "Content Playlist",
                "Course",
                "Digital Textbook",
                "Question paper"
            ],
            "contentPrimaryCategories": [
                "Course Assessment",
                "eTextbook",
                "Explanation Content",
                "Learning Resource",
                "Practice Question Set",
                "Teacher Resource",
                "Exam Question"
            ],
            "name": "Sunbird Org",
            "primaryCategories": [],
            "additionalCategories": [],
            "lastUpdatedOn": "2023-05-26T07:53:39.709+0000",
            "defaultCourseFramework": "TPD",
            "collectionAdditionalCategories": [
                "Textbook",
                "Lesson Plan"
            ],
            "assetPrimaryCategories": [
                "Asset",
                "CertAsset",
                "Certificate Template"
            ],
            "contentAdditionalCategories": [
                "Classroom Teaching Video",
                "Concept Map",
                "Curiosity Question Set",
                "Experiential Resource",
                "Explanation Video",
                "Focus Spot",
                "Learning Outcome Definition",
                "Lesson Plan",
                "Marking Scheme Rubric",
                "Pedagogy Flow",
                "Previous Board Exam Papers",
                "TV Lesson",
                "Textbook"
            ],
            "status": "Live",
            "defaultFramework": "NCF"
        }
    }
}
```

Because of the above File upload editor is using default primary category of `Learning Resource` while creating the content.

[2023-05-29T04:53:52.040] [ERROR] api - {"id":"e29924f6-fd78-fca8-9a1d-6ed9e84f1d7a","msg":"Getting error from content provider while creating content","err":{"err":null,"errCode":"404","errMsg":"Entry is not found in cassandra for content with identifier: obj-cat:learning-resource_content_all {}","responseCode":"RESOURCE_NOT_FOUND"},"additionalInfo":{"ekStepReqData":{"request":{"content":{"name":"VS29","description":"Enter description for Resource","createdBy":"4f2bcb07-6d8e-4022-9866-6e54f1222514","organisation":["Sunbird Org"],"createdFor":["0137972723324682240"],"framework":"sunbird_k-12","mimeType":"application/vnd.ekstep.ecml-archive","resourceType":"Learn","creator":"SunbirdOrgAdmin4","contentType":"Resource","code":"org.sunbird.gqwJ0g"}}}}}

[@maheshkumargangula](https://github.com/maheshkumargangula), [@vrayulu](https://github.com/vrayulu), @Krishnaj20,[@pallakartheekreddy](https://github.com/pallakartheekreddy), [@gandham-santhosh](https://github.com/gandham-santhosh), [@rajeevsathish](https://github.com/rajeevsathish)

**Comment by pallakartheekreddy** [comment]
@vinukumar-vs Currently, Channel API behaviour is if there are no primary categories created in the system, it gets the categories from stem-level(content-service level) configurations. Because of this reason, you are seeing above values.

**Reply by pallakartheekreddy** [reply]
@vinukumar-vs Hope your query is now resolved. Please mark the thread as answered

**Reply by vinukumar-vs** [reply]
This is answered my query, but it will be an issue for new adopters. They have to configure this primary category as must.

**Reply by Krishnaj20** [reply]
@pallakartheekreddy - what are the next steps on this thread?

**Reply by pallakartheekreddy** [reply]
@vinukumar-vs  The configurability of these values, which are considered primary categories, operates at the system level. During the installation of a content service, it's possible to specify which primary categories should be included. The Channel API will then exclusively consider and provide these specified categories in its responses. 

So, it's not mandatary adopter to use the same primary categories. In cases where an adopter does not make any specific configurations during installation, the default values will be automatically included. 

@Krishnaj20

**Reply by vinukumar-vs** [reply]
@pallakartheekreddy As we discussed, if it is mentioned anywhere in the installation document or microsite, it will be good for any adopters to change the values while setting-up the content service. 
I didnt find any documentation taking about content-service configuration. It will be good if you document this as part of content-service installation.

@Krishnaj20, [@vrayulu](https://github.com/vrayulu), @AmiableAnil, [@gandham-santhosh](https://github.com/gandham-santhosh), [@maheshkumargangula](https://github.com/maheshkumargangula)
